lib/logdna/client.rb

```diff
@@ -53,23 +53,18 @@ def schedule_flush
     end
 
     def write_to_buffer(msg, opts)
-      if @lock.try_lock
+      @lock.synchronize do
         processed_message = process_message(msg, opts)
         new_message_size = processed_message.to_s.bytesize
         @buffer.push(processed_message)
         @buffer_byte_size += new_message_size
-        @flush_scheduled = true
-        @lock.unlock
 
         if @flush_limit <= @buffer_byte_size
-          flush
+          Thread.new { flush }
```

Comment:
Constantly creating threads will incur a performance cost, can we consider using a thread pool?

Also, do we plan on trying to stop the scheduled flush when a manual flush occurs? If we have a handle for the scheduled flush this might be able to help: https://ruby-doc.org/core-2.5.0/Thread.html#class-Thread-label-Thread+status